documentation/customizing-valtimo/front-end-customization/custom-components/custom-ui-component-process-link.md

```
@@ -0,0 +1,53 @@
+  documentId: string | null;
+  submittedEvent: EventEmitter<any>;
+}
+```
+
+The component receives `taskInstanceId` (for user tasks), `processDefinitionKey`, `documentDefinitionName`, and `documentId` as inputs, and must emit `submittedEvent` when the user completes the form.
+
+For user tasks, the custom component is responsible for completing the task on the backend (e.g. by calling `TaskService.completeTask()`). After emitting `submittedEvent`, the platform will show a success toast, close the modal, and refresh the task list.
+
+Register the component in your app module using the `FORM_CUSTOM_COMPONENT_TOKEN`:
+
+#### **`app.module.ts`**
```

markdownlint-cli2 (0.22.1)

[warning] Heading levels should only increment by one level at a time
Expected: h2; Actual: h4

(MD001, heading-increment)